feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/ZoomBarComponents.kt

```diff
@@ -0,0 +1,226 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.google.jetpackcamera.feature.preview.ZoomControlUiState
+import java.math.RoundingMode
+import java.text.DecimalFormat
+
+@Composable
+fun ZoomButtonRow(
+    modifier: Modifier = Modifier,
+    zoomControlUiState: ZoomControlUiState.Enabled,
+    buttonSize: Dp = 55.dp,
+    spacing: Dp = 8.dp,
+    onChangeZoom: (Float) -> Unit
+) {
+    val selectedOptionIndex: Int by remember(zoomControlUiState) {
+        // if animating towards a value, then that option will be selected
+        // otherwise, select the closest option that is less than the current zoom ratio
+        val checkValue =
+            zoomControlUiState.animatingToValue ?: zoomControlUiState.primaryZoomRatio ?: 1f
+        derivedStateOf {
+            if (checkValue >= 1f) {
+                // -1 if no index is found
+                zoomControlUiState.zoomLevels.indexOfLast { zoomLevelOption ->
+                    checkValue >= zoomLevelOption
+                }
+            } else {
+                0
+            }
+        }
+    }
+
+    Box(
+        modifier = modifier
+            .background(
+                color = Color.Black.copy(alpha = 0.32f),
+                shape = CircleShape
+            )
+    ) {
+        Row(
+            modifier = Modifier
+                //    .padding(horizontal = spacing)
+                .height(intrinsicSize = IntrinsicSize.Min),
+            horizontalArrangement = Arrangement.spacedBy(spacing),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            zoomControlUiState.zoomLevels.forEachIndexed { index, value ->
+                Box(
+                    modifier = Modifier.width(buttonSize + spacing),
+                    contentAlignment = Alignment.Center
+                ) {
+                    ZoomButton(
+                        targetZoom = value,
+                        currentZoomRatio = { -> (zoomControlUiState.primaryZoomRatio ?: 1f) },
+                        isSelected = selectedOptionIndex == index,
+                        onChangeZoom = onChangeZoom
+                    )
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3ExpressiveApi::class)
+@Composable
+private fun ZoomButton(
+    modifier: Modifier = Modifier,
+    buttonSize: Dp = ButtonDefaults.ExtraSmallContainerHeight,
+    targetZoom: Float,
+    currentZoomRatio: () -> Float,
+    isSelected: Boolean = false,
+    onChangeZoom: (Float) -> Unit
+) {
+    val selectedFormat = DecimalFormat("#.0")
```

Comment:
In the UI mocks we have, when the zoom is settled, there shoulnd't be a trailing "0". Maybe we need to handle that case in a special way?